### DISCUSSION TITLE
Problem with deploy GLSP project templates on Eclipse IDE

### DISCUSSION THREAD

**Discussion by Avaritia1** [eclipse-glsp/glsp#972] [Q&A]
When I recently tried to deploy the GLSP [project template](https://github.com/eclipse-glsp/glsp-examples/tree/master/project-templates/java-emf-eclipse) in Eclipse IDE, I found that the plug-ins are no longer available. Everything worked fine before that. Can someone help with this? Any pointers are appreciated.
![image](https://user-images.githubusercontent.com/80332537/230771677-43ea6d2c-9d5d-481c-b4f2-6eedecb9e680.png)

**Comment by tortmayr** [comment]
Hi @Avaritia1,
it looks like one of the p2 update sites that we use in the target platform is no longer available.
Apparently  "http://dl.maketechnology.io/chromium-cef/rls/repository" does no longer exist.

If you simply remove this dependency from the target you should be able to build again. Instead of chromium the Eclipse Integration than will use a platform specific browser (edge for Windows, cocoa/gtk for osx/linux). This should work just fine as well and will be the new default for GLSP anyways.

Nevertheless I have created https://github.com/eclipse-glsp/glsp/issues/973 to track this and fix the project template.

**Reply by Avaritia1** [reply]
Thanks for the help! It works again

**Comment by srcnuzn** [comment]
Hey @tortmayr,

with your fix in https://github.com/eclipse-glsp/glsp/issues/973 I can now successfully  execute `yarn build` and even launch `TaskListEditor.launch` without modifying the target. But after importing the `TaskLiskExample` and double clicking on `example.tasklist` the editor opens but stays blank. In my debugging console I get the follownig error:
```
2023-05-18 18:07:33.522:INFO::main: Logging initialized @220172ms to org.eclipse.jetty.util.log.StdErrLog
2023-05-18 18:07:33.804:INFO:oejs.Server:main: jetty-9.4.44.v20210927; built: 2021-11-04T08:22:36.864Z; git: 8da83308eeca865e495e53ef315a249d63ba9332; jvm 19.0.1+10
2023-05-18 18:07:33.834:INFO:oejs.session:main: DefaultSessionIdManager workerName=node0
2023-05-18 18:07:33.834:INFO:oejs.session:main: No SessionScavenger set, using defaults
2023-05-18 18:07:33.835:INFO:oejs.session:main: node0 Scavenging every 660000ms
2023-05-18 18:07:33.852:INFO:oejsh.ContextHandler:main: Started o.e.j.s.ServletContextHandler@4f1072ee{/,null,AVAILABLE}
2023-05-18 18:07:33.893:INFO:oejs.AbstractConnector:main: Started ServerConnector@1e834a5b{HTTP/1.1, (http/1.1)}{localhost:52059}
2023-05-18 18:07:33.894:INFO:oejs.Server:main: Started @220544ms
SWT.CHROMIUM style was used but chromium.swt win32 (or CEF binaries) fragment/jar is missing.
```
Is this related to your change or is it a separate issue?

**Reply by srcnuzn** [reply]
The log message with `SWT.CHROMIUM` occurs when I'm in `GLSPDiagramEditor.createBrowser` which is triggered after double clicking on `example.tasklist`. I guess there is an incompatible version?

[GLSPDiagramEditor (v1.0.0)](https://github.com/eclipse-glsp/glsp-eclipse-integration/blob/v1.0.0/server/plugins/org.eclipse.glsp.ide.editor/src/org/eclipse/glsp/ide/editor/ui/GLSPDiagramEditor.java#L345)  (used by the template)
```java
protected Browser createBrowser(final Composite parent) {
      Browser browser = new FocusAwareBrowser(parent, SWT.NO_SCROLL | SWT.CHROMIUM);
      browser.setLayoutData(new GridData(GridData.FILL, GridData.FILL, true, true));
      toDispose.add(browser::dispose);
      return browser;
}
```
vs.

[GLSPDiagramEditor (v1.1.0.RC08)](https://github.com/eclipse-glsp/glsp-eclipse-integration/blob/v1.1.0.RC08/server/plugins/org.eclipse.glsp.ide.editor/src/org/eclipse/glsp/ide/editor/ui/GLSPDiagramEditor.java#L364)
```java
protected Browser createBrowser(final Composite parent) {
      Browser browser = new FocusAwareBrowser(parent, SWT.NO_SCROLL | SWT.EDGE);
      browser.setLayoutData(new GridData(GridData.FILL, GridData.FILL, true, true));
      toDispose.add(browser::dispose);
      return browser;
}
```

**Reply by tortmayr** [reply]
Good catch!. Yes that was indeed a  version conflict.
I have implemented a workaround for the Eclipse-Project-Template (https://github.com/eclipse-glsp/glsp-examples/pull/187) that should fix the issue under Windows

**Reply by srcnuzn** [reply]
Hi @tortmayr , thanks for the fixes. Due to vacation I was only now able to test it. 
The emf-java example seems to work on the latest master.
Is this how the example task-list editor is supposed to look like?

![image](https://github.com/eclipse-glsp/glsp/assets/61967763/31ba4130-b6ee-4ef8-8a5a-bbfde57ced97)

**Reply by tortmayr** [reply]
Looks fine to me. The task list editor is a very minimal example that in a nutshell just has a couple of resizable nodes.
